pandas/core/internals.py

```diff
@@ -3597,7 +3597,14 @@ def value_getitem(placement):
                 return value
         else:
             if value.ndim == self.ndim - 1:
-                value = value.reshape((1,) + value.shape)
+                # gh-13012: 'reshape' is deprecated for Categorical
+                # and Series and will be removed in a subsequent version,
+                # hence the two checks done before calling 'reshape'
+                if isinstance(value, ABCSeries):
+                    value = value._values
```

Comment:
Though why `pandas.types.concat`?  We aren't really concatenating anything here.  My first inclination was to try to put in `pandas.compat`, but I don't see a good home for it there.